bundles/org.openhab.binding.juicenet/src/main/resources/OH-INF/thing/juicenet-device.xml

```diff
@@ -0,0 +1,271 @@
+				<option value="disconnect">Disconnected</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="overrideType">
+		<item-type>Switch</item-type>
+		<label>Override State</label>
+		<description>Smart charging is overridden.</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="chargingTimeLeftType">
+		<item-type>Number:Time</item-type>
+		<label>Charging Time Left</label>
+		<description>Charging time left (seconds).</description>
+		<state readOnly="true" pattern="%.0f min"/>
+	</channel-type>
+
+	<channel-type id="plugUnplugTimeType">
+		<item-type>DateTime</item-type>
+		<label>Plug/Unplug Time</label>
+		<description>Last time of either plug-in or plug-out.</description>
+		<state readOnly="true" pattern="%1$tB %1$te, %1$tY %1$tl:%1$tM %1$tp"/>
+	</channel-type>
+
+	<channel-type id="targetTimeType">
+		<item-type>DateTime</item-type>
+		<label>Target Time</label>
+		<description>“Start charging” start time, or time to start when overriding smart charging.</description>
+		<state pattern="%1$tB %1$te, %1$tY %1$tl:%1$tM %1$tp"/>
+	</channel-type>
+
+	<channel-type id="unitTimeType">
+		<item-type>DateTime</item-type>
+		<label>Unit Time</label>
+		<description>Current time on the unit.</description>
+		<state readOnly="true" pattern="%1$tB %1$te, %1$tY %1$tl:%1$tM %1$tp"/>
+	</channel-type>
+
+	<channel-type id="temperatureType">
+		<item-type>Number:Temperature</item-type>
+		<label>Temperature</label>
+		<description>Current temperature at the unit.</description>
+		<state readOnly="true" pattern="%.0f %unit%"/>
+	</channel-type>
+
+	<channel-type id="ampsLimitType">
+		<item-type>Number:ElectricCurrent</item-type>
+		<label>Current Limit</label>
+		<description>Max charging current allowed. (A)</description>
+		<state pattern="%d A"/>
+	</channel-type>
+
+	<channel-type id="ampsCurrentType">
+		<item-type>Number:ElectricCurrent</item-type>
+		<label>Current</label>
+		<description>Current charging current. (A)</description>
+		<state readOnly="true" pattern="%.1f A"/>
+	</channel-type>
+
+	<channel-type id="voltageType">
+		<item-type>Number:ElectricPotential</item-type>
+		<label>Voltage</label>
+		<description>Current voltage. (V)</description>
+		<state readOnly="true" pattern="%d V"/>
+	</channel-type>
+
+	<channel-type id="whEnergyType">
+		<item-type>Number:Energy</item-type>
+		<label>Current Energy</label>
+		<description>Current power level of vehicle. (Wh)</description>
+		<state readOnly="true" pattern="%.1f kWh"/>
+	</channel-type>
+
+	<channel-type id="savingsType">
+		<item-type>Number</item-type>
+		<label>Savings</label>
+		<description>Current session EV savings.</description>
+		<state readOnly="true" pattern="$%.2f"/>
+	</channel-type>
+
+	<channel-type id="wattPowerType">
+		<item-type>Number:Power</item-type>
+		<label>Power</label>
+		<description>Current charging power. (W)</description>
+		<state readOnly="true" pattern="%.2f kW"/>
```

Comment:
There is some inconsistency between description and supplied unit (W vs kW). Best would probably be to drop "W" from the description and replace "kW" with "%unit%" as unit will be supplied when creating the `QuantityType`:
```suggestion
		<description>Current charging power.</description>
		<state readOnly="true" pattern="%.2f %unit%"/>
```

Additionally, reconsider the name of the channel type. Perhaps "chargingPower"?

Some similar comments about unit for other channel type definitions, I will go through them later.

Reply:
That makes sense.  I was torn between adopting the names used in the JuiceNet API vs. coming up with better names.   Related, there are several quantityTypes which don't exists (such as EnergyPerMile o ro miles per gallon).  Those i have just kept as Decimal values.

Reply:
Regarding names used in the API, don't feel bound by them. This binding is an abstraction from that API, so that users can have a unified experience interacting with various equipment implementing completely different API's.